addons/sale_timesheet/models/sale_order.py

```diff
@@ -136,8 +136,35 @@ def action_view_timesheet(self):
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    qty_delivered_method = fields.Selection(selection_add=[('timesheet', 'Timesheets')])
     task_id = fields.Many2one('project.task', 'Task', help="Task generated by the sales order item")
     is_service = fields.Boolean("Is a Service", compute='_compute_is_service', store=True, help="Sales Order item should generate a task and/or a project, depending on the product settings.")
+    analytic_line_ids = fields.One2many(domain=[('project_id', '=', False)])  # only analytic lines, not timesheets (since this field determine if SO line came from expense)
+
+    @api.multi
+    @api.depends('product_id.type', 'product_id.service_type')
+    def _compute_qty_delivered_method(self):
+        """ Sale Timesheet module compute delivered qty for product [('type', 'in', ['service']), ('service_type', '=', 'timesheet')] """
+        super(SaleOrderLine, self)._compute_qty_delivered_method()
+        for line in self:
+            if not line.is_expense and line.product_id.type == 'service' and line.product_id.service_type == 'timesheet':
+                line.qty_delivered_method = 'timesheet'
+
+    @api.multi
+    @api.depends('analytic_line_ids.project_id')
+    def _compute_qty_delivered(self):
+        super(SaleOrderLine, self)._compute_qty_delivered()
+
+        lines_by_timesheet = self.filtered(lambda sol: sol.qty_delivered_method == 'timesheet')
+        domain = lines_by_timesheet._timesheet_compute_delivered_quantity_domain()
+        mapping = lines_by_timesheet._get_delivered_quantity_by_analytic(domain)
+        for line in lines_by_timesheet:
+            line.qty_delivered = mapping.get(line.id, 0.0)
+
+    @api.multi
```

Comment:
This is also a side-remark: decorating private methods is not necessary.

Reply:
You sounds like xmo ... :p 
I know this is true, but migrating all the base code impacted my brain in some aspects :face_with_head_bandage: